README.md

```diff
@@ -65,44 +59,171 @@ make submit-flu
+  script that polls your source frequently, gets the raw data, and builds an
+  archive. See `scripts/build_nhsn_archive.R` for an example of how to do this.
+  Then, you can make your forecasting pipeline depend on this archive, which
+  (assuming everything went well) will be up to date.
+
+### Running and debugging pipelines
+
+You can run these pipelines as described above, but you can also run them in the R REPL.
+
+```r
+suppressPackageStartupMessages(source("R/load_all.R"))
+
+# Make sure to set the project to the one you want to run
+Sys.setenv(TAR_PROJECT = "covid_hosp_explore")
+
+# Run the pipeline
+tar_make()
+```
 
-Insert a `browser()` statements into the code you want to debug and then run
+Frequently, you will want to run and debug a single target. First place a
+`browser()` statement in the target function. Then run the following command:
 
 ```r
-# use_crew=FALSE disables parallelization
 tar_make(target_name, callr_function = NULL, use_crew = FALSE)
 ```
 
-### Pipeline Design
+This will run the pipeline up to and including the target you specified and drop
+you into the R debugger.
 
-See [this diagram](https://excalidraw.com/#json=AmMzzAKxSF5rz1dvuDJxj,0b53_5Ro6xwm13uQXrIGMQ).
-Double diamond objects represent "plates" (to evoke [plate notation](https://en.wikipedia.org/wiki/Plate_notation), but don't take the comparison too literally), which are used to represent multiple objects of the same type (e.g. different forecasters).
+### Adding a new forecaster
 
-## Notes on Forecaster Types
+To add a new forecaster, we recommend the following process.
 
-### Basic
+First, add a new forecaster function in `R/forecasters/` with the following
+general format:
 
-The basic forecaster takes in an epi_df, does some pre-processing, does an epipredict workflow, and then some post-processing
+```r
+function(epi_data,
+         outcome,
+         ahead = 1,
+         ...)
+```
 
-### Ensemble
+The `epi_data` argument is the data frame of epidemic data. You can typically
+assume that this is an `epi_df` object (with all the standard columns, like
+`geo_value`, `time_value`, `version`, etc.). The `outcome` argument is the
+column name of the outcome variable that you want to forecast. The `ahead`
+argument is the number of time steps ahead to forecast. The `...` argument is
+for additional arguments that are passed to the forecaster.
+
+Some other common arguments are:
+
+- `extra_sources`: a string of column names of additional data to use for the
+  forecast. This is often empty, but useful if you have exogenous variables.
+- `filter_source`: a string of the source to filter to. This is often empty, but
+  sometimes it's a string like "nhsn". This allows us to create one large
+  dataset (such as `joined_archive_data`) that contains all the data for all
+  sources, which we can feed uniformly to every forecaster and have them select
+  which data they want to use.
```

Comment:
I think this ends up being required if you want to run it on flu.

In general, it might be easier to point to a template forecaster to some degree that's what `scaled_pop` is for, though we could probably go more minimal.